cmd/generate/generate.go

```diff
@@ -290,6 +297,26 @@ func generateSS(o *GenerateOptions, deployment types.Deployment) (*types.ComMatr
 	if err != nil {
 		return nil, fmt.Errorf("failed to create namespace: %v", err)
 	}
+	// Always delete namespace and wait for full removal to avoid Terminating races on reruns
+	defer func() {
+		if delErr := o.utilsHelpers.DeleteNamespace(consts.DefaultDebugNamespace); delErr != nil {
+			log.Warnf("failed to delete namespace %s: %v", consts.DefaultDebugNamespace, delErr)
+			return
+		}
+		if pollErr := wait.PollUntilContextTimeout(context.TODO(), time.Second, 2*time.Minute, true, func(ctx context.Context) (bool, error) {
+			ns := &corev1.Namespace{}
+			err := o.cs.Get(context.TODO(), ctrlclient.ObjectKey{Name: consts.DefaultDebugNamespace}, ns)
+			if apierrors.IsNotFound(err) {
+				return true, nil
+			}
+			if err != nil {
+				return false, nil // keep retrying
```

Comment:
but I think we should at least print a long with the error message if not it will get lost